src/Castle.Windsor.Extensions.DependencyInjection/Scope/ExtensionContainerRootScopeAccessor.cs

```diff
@@ -15,24 +15,29 @@
 namespace Castle.Windsor.Extensions.DependencyInjection.Scope
 {
 	using System;
-	
+
 	using Castle.MicroKernel.Context;
 	using Castle.MicroKernel.Lifestyle.Scoped;
 
 	internal class ExtensionContainerRootScopeAccessor : IScopeAccessor
 	{
 		public ILifetimeScope GetScope(CreationContext context)
 		{
-			if (ExtensionContainerRootScope.RootScope == null)
+			if (ExtensionContainerScope.Current == null)
+			{
+				throw new InvalidOperationException("No root scope");
+			}
+
+			if (ExtensionContainerScope.Current.RootScope == null)
 			{
 				throw new InvalidOperationException("No root scope");
 			}
 
-			return ExtensionContainerRootScope.RootScope;
+			return ExtensionContainerScope.Current.RootScope;
 		}
```

Comment:
This function is probably better written like below, or maybe even more concise, at minimum combine the duplication of if-statements with the same body:

```csharp
public ILifetimeScope GetScope(CreationContext context)
{
    return ExtensionContainerScope.Current?.RootScope ??
        throw new InvalidOperationException("No root scope");
}
```

Reply:
fixed